va/va_trace.h

```diff
@@ -54,6 +57,171 @@ extern int va_trace_flag;
         va_TraceStatus(dpy, __func__, ret);     \
     }
 
+/** \brief event id definition
+ * identifier of trace event, coresponding the task value in the manifest, located in libva-util/tracetool
+ * the trace tool will translate this id to event name, also the trace data carried along. 
+ * Note: make sure eventi id definition backward compatible */
+enum {
+    INVALIDE_ID = 0,
+    CREATE_CONFIG = 1,
+    DESTROY_CONFIG,
+    CREATE_CONTEXT,
+    DESTROY_CONTEXT,
+    CREATE_BUFFER,
+    DESTROY_BUFFER,
+    CREATE_SURFACE,
+    DESTROY_SURFACE,
+    BEGIN_PICTURE,
+    RENDER_PICTURE,
+    END_PICTURE,
+    BUFFER_DATA,
+    SYNC_SURFACE,
+    SYNC_SURFACE2,
+    QUERY_SURFACE_ATTR,
+};
+
+/** \brief event opcode definition
+ * identifier of trace event operation, coresponding the opcode value in the manifest.
+ * 4 opcode prefined */
+#define TRACE_INFO  0
+#define TRACE_BEGIN 1
+#define TRACE_END   2
+#define TRACE_DATA  3
+
+/** \brief event data structure
+ * scattered structure to pass event data without copy, each entry contail the event data address and size.
+ * va_TraceEvent will write these raw data into ftrace entry */
+typedef struct va_event_data {
+    void *buf;
+    unsigned int size;
+} VAEventData;
+
+
+/** \brief VA_TRACE
+ * trace interface to send out trace event with empty event data. */
+#define VA_TRACE(dpy,id,op)                         \
+    if (va_trace_flag & VA_TRACE_FLAG_FTRACE) {     \
+        va_TraceEvent(dpy, id, op, 0, NULL);        \
+    }
+/** \brief VA_TRACE_V
+ * trace interface to send out trace event with 1 data element from variable. the variable data type could be 8/16/32/64 bitsize */
+#define VA_TRACE_V(dpy,id,op,v)                     \
```

Comment:
already have if (xx) {} to enclose code block, should have same effect with do {} while.
otherwise the code would be like:
    do {
        if (va_trace_flag & VA_TRACE_FLAG_FTRACE) {     \
            VAEventData desc[1] = {{&v, sizeof(v)}};    \
            va_TraceEvent(dpy, id, op, 1, desc);        \
        }
    } while(0)
what do you think?

Reply:
No, it wont' have same effect. Consider:
```
if (a) VA_TRACE()
else print("\n");
```

With you current code else will be wrongly interpreted as the one belonging to `if()` inside macro.

Reply:
ok, will update